src/backend/cuda/kernel/scan_by_key/CMakeLists.txt

```diff
@@ -14,32 +14,15 @@ foreach(STR ${FILESTRINGS})
     endif()
 endforeach()
 
-cuda_add_cuda_include_once()
-
 foreach(SBK_BINARY_OP ${SBK_BINARY_OPS})
-    # When using cuda_compile with older versions of FindCUDA. The generated targets
-    # have the same names as the source file. Since we are using the same file for
-    # the compilation of these targets we need to rename them before sending them
-    # to the cuda_compile command so that it doesn't generate multiple targets with
-    # the same name
-    file(COPY "${CMAKE_CURRENT_SOURCE_DIR}/kernel/scan_by_key/scan_by_key_impl.cu"
-      DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/kernel/scan_by_key")
-    file(RENAME "${CMAKE_CURRENT_BINARY_DIR}/kernel/scan_by_key/scan_by_key_impl.cu"
-      "${CMAKE_CURRENT_BINARY_DIR}/kernel/scan_by_key/scan_by_key_impl_${SBK_BINARY_OP}.cu")
-
-    cuda_compile(scan_by_key_gen_files "${CMAKE_CURRENT_BINARY_DIR}/kernel/scan_by_key/scan_by_key_impl_${SBK_BINARY_OP}.cu"
-          "${CMAKE_CURRENT_SOURCE_DIR}/kernel/scan_dim_by_key_impl.hpp"
-          "${CMAKE_CURRENT_SOURCE_DIR}/kernel/scan_first_by_key_impl.hpp"
-          OPTIONS
-          -DSBK_BINARY_OP=${SBK_BINARY_OP} "${platform_flags} ${cuda_cxx_flags} -DAFDLL"
-      )
-
-    list(APPEND SCAN_OBJ ${scan_by_key_gen_files})
-endforeach(SBK_BINARY_OP ${SBK_BINARY_OPS})
+  configure_file(
```

Comment:
Couldn't we just add cuda_scan_by_key target to the static library?

Reply:
I don't understand quite what you mean. The generated cpp files can be directly included into afcuda target and they don't take any time at all since it is going through nvcc anymore. It doesn't make sense to retain this old approach.